pkg/flamingo/ftp.go

```diff
@@ -0,0 +1,122 @@
+package flamingo
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"net"
+	"strings"
+	"sync"
+)
+
+// ConfFTP holds information for a FTP server.
+type ConfFTP struct {
+	BindPort     uint16
+	BindHost     string
+	RecordWriter *RecordWriter
+	shutdown     bool
+	listener     net.Listener
+	m            sync.Mutex
+}
+
+// NewConfFTP creates a default configuration for the FTP capture server.
+func NewConfFTP() *ConfFTP {
+	return &ConfFTP{
+		BindPort: 21,
+		BindHost: "[::]",
+	}
+}
+
+// IsShutdown checks to see if the service is shutting down.
+func (c *ConfFTP) IsShutdown() bool {
+	c.m.Lock()
+	defer c.m.Unlock()
+	return c.shutdown
+}
+
+// Shutdown flags the service to shut down
+func (c *ConfFTP) Shutdown() {
+	c.m.Lock()
+	defer c.m.Unlock()
+	c.listener.Close()
+}
+
+// SpawnFTP creates a new FTP capture server.
+func SpawnFTP(c *ConfFTP) error {
+	listener, err := net.Listen("tcp", fmt.Sprintf("%s:%d", c.BindHost, c.BindPort))
+	if err != nil {
+		return err
+	}
+	c.listener = listener
+	go startFTP(c)
+	return nil
+}
+
+func startFTP(c *ConfFTP) {
```

Comment:
startFTP needs some way to break out of this on shutdown

Reply:
Change `continue` to `break`?